### PR TITLE
Update Columns to return visible ordered column list

### PR DESCRIPTION
**Pull request by RFSH**
Related Issue: #51 

This PR,
- Changes `._columns` in `Columns` class from Object to Array.
- Adds a `.table` to `Columns` class since it is only used to store columns of a single table.
- Adds `.contextualize(context)` to `Columns` class, which contextualizes the object and returns a new `Columns` object based on [its annotation](https://github.com/informatics-isi-edu/ermrest/blob/master/user-doc/annotation.md#2016-visible-columns).
- Adds the following unit tests:

   For determining order of visible columns, 
      - should use the defined order in annotations based on its context.
      - should use `entry` context when context is `edit` or `create` and they are not present in annotation.
      - should use default (`*`) context for any context not matched by a more specific context name.
      - should use the default order when context not matched by a more specific context name and default (`*`) context is not present.
      - should ignore the column names that do not correspond to any column in the table.
      - should ignore repetitive column names and just consider the first one.

### CODE REVIEW

File: js/ermrest.js [hunk 1]
```diff
@@ -1045,15 +1085,64 @@ var ERMrest = (function (module) {
...
+            return this._columns[pos];
+        },
 
+        /***
+         * Get columns based on the context.
+         *
+         * @param {string} context
+         * @returns {ERMrest.Columns} columns with predefined order based on the context.
+         */
+        contextualize: function (context) {
```
Comment (robes) [inline js/ermrest.js]: i believe, this should be a private not public api change.

**Comment by robes** [comment]
Overall, it looks good. But two unexpected changes were made to the _public_ api. Those `.table` and `.contextualize()` ought to be _private_ (i.e., `._...`). After you fix those issues, and resovle the conflicts (due to other's commits) and this is ready to be merged.

**Comment by RFSH** [comment]
@robes But if we make the `.contextualize()` function private, then nobody can call it outside Columns class. How is it going to be called?

**Comment by robes** [comment]
@RFSH private properties can be used internal to the ERMrest module itself. To clarify, if something is declared pseudo-private by convention "_abc" it means we are telling our public clients of the API, "don't use that". Its okay if _we_ use it internal to the api. It doesn't have to be internal to a class, it can be used by us across our sub-modules.

**Comment by RFSH** [comment]
@robes `contextualize()` and `table` are now private. Conflicts with master are resolved too.

**Pull request merged**